tianshou/data/utils.py

```diff
@@ -37,6 +34,16 @@ def to_torch(x: Union[torch.Tensor, dict, Batch, np.ndarray],
             x[k] = to_torch(v, dtype, device)
     elif isinstance(x, Batch):
         x.to_torch(dtype, device)
+    elif isinstance(x, (np.number, np.bool_, Number)):
+        x = to_torch(np.asanyarray(x), dtype, device)
+    elif isinstance(x, list) and len(x) > 0 and \
+            isinstance(x[0], (np.number, np.bool_, Number)):
```

Comment:
Why do you use `isinstance(x, (np.number, np.bool_, Number))`?
Can we use the np array conversion and let numpy automatically determine the type? For example,
```python
if not isinstance(x, (torch.Tensor, np.ndarray)):
    x = np.asanyarray(x)
# then check the type
...
```

Reply:
No, since `torch.Tensor` only supports the types `(np.number, np.bool_, Number)` :smile: You can check by yourself:
```
torch.tensor(np.asanyarray(['nan', 'er']))
>>> TypeError: can't convert np.ndarray of type numpy.str_. The only supported types are: float64, float32, float16, int64, int32, int16, int8, uint8, and bool.
```